extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,122 @@
+    internal class BlobStorageConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is not IBindingData bindingData)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            if (bindingData.Source is not "Microsoft.Azure.WebJobs.Extensions.Storage.Blobs")
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var content = bindingData.Content.ToObjectFromJson<Dictionary<string, string>>();
+
+            content.TryGetValue("Connection", out var connectionName);
+            content.TryGetValue("BlobName", out var blobName);
+            content.TryGetValue("ContainerName", out var containerName);
+
+            if (string.IsNullOrEmpty(connectionName)
+                || string.IsNullOrEmpty(blobName)
+                || string.IsNullOrEmpty(containerName))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var connectionString = connectionName is null ? null : Environment.GetEnvironmentVariable(connectionName);
+            object result = ToTargetType(context.TargetType, connectionString, containerName, blobName);
+
+            if (result is not null)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Success(result));
+            }
+
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+        private object? ToTargetType(Type targetType, string connectionString, string containerName, string blobName) => targetType switch
+        {
+            Type _ when targetType == typeof(String)            => GetBlobString(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Stream)            => GetBlobStream(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Byte[])            => GetBlobBinaryData(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient)        => CreateBlobReference<BlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlockBlobClient)   => CreateBlobReference<BlockBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient)    => CreateBlobReference<PageBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient)  => CreateBlobReference<AppendBlobClient>(connectionString, containerName, blobName),
+            _ => CreateTargetObject(targetType, connectionString, containerName, blobName)
+        };
+
+        private object CreateTargetObject(Type targetType, string connectionString, string containerName, string blobName)
```

Comment:
So CreateTargetObject is meant to be a catch all for POCOs and other types we don't recognize? E.g. in proc, we support binding a Blob to a TextReader. If they try to use that same code out of proc, you'll end up trying to deserialize into a TextReader?

Reply:
Yeah I don't think we're planning to support TextReader, will try and get list of those together so we can deal with them